internal/component/database_observability/postgres/collector/schema_details.go

```diff
@@ -205,6 +206,72 @@ type foreignKey struct {
 	ReferencedColumnName string `json:"referenced_column_name"`
 }
 
+// TableRegistry is a source-of-truth cache that keeps track of databases, schemas, tables
+type TableRegistry struct {
+	mu     sync.RWMutex
+	tables map[string]map[string]map[string]bool // map[database]map[schema]map[table]bool
```

Comment:
Can you create alias types if it would help for readability on what this represents